ts/modules/marker-clusters.ts

```diff
@@ -0,0 +1,1948 @@
+/* *
+ *
+ *  Marker clusters module.
+ *
+ *  (c) 2010-2019 Torstein Honsi
+ *
+ *  Author: Wojciech Chmiel
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+'use strict';
+
+import H from './../parts/Globals.js';
+import U from './../parts/Utilities.js';
+
+/**
+ * Internal types
+ * @private
+ */
+declare global {
+    namespace Highcharts {
+        type ClusterAlgorithms = (
+            'grid'|'kmeans'|'optimizedKmeans'
+        );
+        interface LayoutAlgorithmOptions {
+            type: (ClusterAlgorithms | Function);
+            gridSize: (number | string);
+            processedGridSize: number;
+            distance: (number | string);
+            processedDistance: number;
+            iterations?: number;
+            kmeansThreshold?: number;
+        }
+        interface StyleOptions {
+            symbol: string;
+            fillColor: ColorType;
+            lineColor: ColorType;
+            lineWidth: number;
+            radius: number;
+        }
+        interface GridOffsetObject {
+            plotLeft: number;
+            plotTop: number;
+        }
+        interface SortedKmeansClusterObject {
+            clusterIndex: number;
+            distance: number;
+        }
+        interface RealExtremesObject {
+            minX: number;
+            maxX: number;
+            minY: number;
+            maxY: number;
+        }
+        interface ClusterZonesOptions {
+            from: number;
+            to: number;
+            style: StyleOptions;
+            zoneIndex: number;
+            className?: string;
+        }
+        interface OnDrillToClusterCallbackFunction {
+            (this: Point, event: PointClickEventObject): void;
+        }
+        interface ClusterEventsDictionary {
+            onDrillToCluster: OnDrillToClusterCallbackFunction;
+        }
+        interface MarkerClusterOptions {
+            enabled?: boolean;
+            allowOverlap?: boolean;
+            minimumClusterSize?: number;
+            drillToCluster?: boolean;
+            layoutAlgorithm: LayoutAlgorithmOptions;
+            style?: StyleOptions;
+            dataLabels?: DataLabelsOptionsObject;
+            zones?: Array<ClusterZonesOptions>;
+            states?: PointStatesOptionsObject;
+            events?: ClusterEventsDictionary;
+        }
+        interface PointMarkerOptionsObject {
+            cluster?: MarkerClusterOptions;
+        }
+        interface PointOptionsObject {
+            lat?: number;
+            lon?: number;
+        }
+        interface SplittedDataObject {
```

Comment:
Should be `SplitDataObject`. "Splitted" is not a valid conjugation. https://www.usingenglish.com/reference/irregular-verbs/split.html.